File: src/rngGame/stats/Use.java
package rngGame.stats;

public class Use extends Item {

	private int hp;

	public Use(Rarity itemRarity) {
		super("Use", itemRarity.getTextureName(), itemRarity);
		hp = switch (getRarity()) {
			case COMMON -> 10;
			case UNCOMMON -> 20;
			case RARE -> 40;
			case VERY_RARE -> 50;
			case EPIC -> 70;
			case LEGENDARY -> 90;
			case GOD -> 130;
			case VOID -> 180;
		};
	}

	public int getHp() {
		return hp;
	}

	public void setHp(int hp) {
		this.hp = hp;
	}

}

File: src/rngGame/stats/Use.kt
package rngGame.stats

class Use(itemRarity: Rarity): Item("Use", itemRarity.textureName, itemRarity) {

	var hp = when(rarity) {
		Rarity.COMMON -> 10
		Rarity.UNCOMMON -> 20
		Rarity.RARE -> 40
		Rarity.VERY_RARE -> 50
		Rarity.EPIC -> 70
		Rarity.LEGENDARY -> 90
		Rarity.GOD -> 130
		Rarity.VOID -> 180
	}

}
